packages/react/src/components/Loader/Loader.tsx

```diff
@@ -94,6 +97,27 @@ class Loader extends UIComponent<WithAsProp<LoaderProps>, LoaderState> {
 
     this.state = {
       visible: this.props.delay === 0,
+      labelLoaderId: getOrGenerateIdFromShorthand('loader-label-', props.label, ''),
+    }
+  }
+
+  getIdFromShorthand = <P extends Record<string, any>>(value: ShorthandValue<P>): string | undefined => {
+    if (_.isNil(value)) {
+      return undefined
+    }
+    let result: string = ''
+    if (React.isValidElement(value)) {
+      result = (value as React.ReactElement<{ id?: string }>).props.id
+    } else if (_.isPlainObject(value)) {
+      result = (value as Record<string, any>).id
+    }
+    
+    return result
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.getIdFromShorthand(this.props.label) !== this.getIdFromShorthand(prevProps.label)) {
+      this.setState(prevState => ({ labelLoaderId: getOrGenerateIdFromShorthand('loader-label-', this.props.label, prevState.labelLoaderId) }))
     }
```

Comment:
Why don't use `getDerivedStateFromProps()`? This will produce an additional render

Reply:
agree - please see [Alert.tsx](https://github.com/microsoft/fluent-ui-react/blob/packages/react/src/components/Alert/Alert.tsx#L157)

Reply:
`getDerivedStateFromProps` should be enough. This is not Autocontrolled component.